build/scripts/patch-gen.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# Copyright 2024 The Cockroach Authors.
+#
+# Use of this software is governed by the CockroachDB Software License
+# included in the /LICENSE file.
+
+set -euo pipefail
```

Comment:
The script is fine. It does seem like overkill to essentially do `git diff > file` though.

Reply:
Right, I added it just for "symmetry" with the apply one. It does save you a little bit in that you don't have to worry about where the patch goes.